### ISSUE TITLE
problem installing classic addons in retail mode

### ISSUE THREAD

**Issue by torkus** [ogri-la/wowman#55]
while attempting to replicate the new problem here: https://github.com/ogri-la/wowman/issues/47

I discovered this:

```clojure
2019-09-20 14:12:31.055 rama WARN [wowman.curseforge-api:64] - no 'retail' release available for 'atlaslootclassic' on curseforge
2019-09-20 14:12:31.073 rama ERROR [wowman.ui.gui:99] -                               java.lang.Thread.run              Thread.java:  748
java.util.concurrent.ThreadPoolExecutor$Worker.run  ThreadPoolExecutor.java:  624
 java.util.concurrent.ThreadPoolExecutor.runWorker  ThreadPoolExecutor.java: 1149
               java.util.concurrent.FutureTask.run          FutureTask.java:  266
                                               ...                               
               clojure.core/binding-conveyor-fn/fn                 core.clj: 2022
                 wowman.ui.gui/async-handler/fn/fn                  gui.clj:   97
                                               ...                               
           orchestra.spec.test/spec-checking-fn/fn                 test.clj:  125
                                               ...                               
      wowman.ui.gui/search-results-install-handler                  gui.clj:  195
                                               ...                               
           orchestra.spec.test/spec-checking-fn/fn                 test.clj:  124
     orchestra.spec.test/spec-checking-fn/conform!                 test.clj:  113
                              clojure.core/ex-info                 core.clj: 4739
clojure.lang.ExceptionInfo: Call to wowman.core/-install-update-these did not conform to spec:
                            nil - failed: map? in: [0 0] at: [:args :updateable-toc-addons :addon?] spec: :wowman.specs/addon-summary
                            nil - failed: map? in: [0 0] at: [:args :updateable-toc-addons :addon?] spec: :wowman.specs/addon
                            nil - failed: map? in: [0 0] at: [:args :updateable-toc-addons :toc-addon?] spec: :wowman.specs/toc
                            nil - failed: map? in: [0 0] at: [:args :updateable-toc-addons :toc-addon?] spec: :wowman.specs/addon-summary
                            nil - failed: map? in: [0 0] at: [:args :updateable-toc-addons :toc-addon?] spec: :wowman.specs/addon
                            nil - failed: map? in: [0 0] at: [:args :updateable-toc-addons :toc-addon?] spec: :wowman.specs/toc-addon
                            
        clojure.spec.alpha/args: ([nil])
     clojure.spec.alpha/failure: :instrument
    clojure.spec.alpha/problems: ({:path [:args :updateable-toc-addons :addon?],
                                   :pred clojure.core/map?,
                                   :val nil,
                                   :via
                                   [:wowman.specs/addon-or-toc-addon
                                    :wowman.specs/addon
                                    :wowman.specs/addon-summary],
                                   :in [0 0]}
                                  {:path [:args :updateable-toc-addons :addon?],
                                   :pred clojure.core/map?,
                                   :val nil,
                                   :via [:wowman.specs/addon-or-toc-addon :wowman.specs/addon],
                                   :in [0 0]}
                                  {:path [:args :updateable-toc-addons :toc-addon?],
                                   :pred clojure.core/map?,
                                   :val nil,
                                   :via
                                   [:wowman.specs/addon-or-toc-addon
                                    :wowman.specs/toc-addon
                                    :wowman.specs/toc],
                                   :in [0 0]}
                                  {:path [:args :updateable-toc-addons :toc-addon?],
                                   :pred clojure.core/map?,
                                   :val nil,
                                   :via
                                   [:wowman.specs/addon-or-toc-addon
                                    :wowman.specs/toc-addon
                                    :wowman.specs/addon
                                    :wowman.specs/addon-summary],
                                   :in [0 0]}
                                  {:path [:args :updateable-toc-addons :toc-addon?],
                                   :pred clojure.core/map?,
                                   :val nil,
                                   :via
                                   [:wowman.specs/addon-or-toc-addon
                                    :wowman.specs/toc-addon
                                    :wowman.specs/addon],
                                   :in [0 0]}
                                  {:path [:args :updateable-toc-addons :toc-addon?],
                                   :pred clojure.core/map?,
                                   :val nil,
                                   :via [:wowman.specs/addon-or-toc-addon :wowman.specs/toc-addon],
                                   :in [0 0]})
        clojure.spec.alpha/spec: #object[clojure.spec.alpha$regex_spec_impl$reify__2509 0x74cfdfc "clojure.spec.alpha$regex_spec_impl$reify__2509@74cfdfc"]
       clojure.spec.alpha/value: ([nil])
     orchestra.spec.test/caller: {:file "gui.clj",
                                  :line 195,
                                  :var-scope wowman.ui.gui/search-results-install-handler}
 unhandled exception in thread
```

1. new addon directory
2. ensure 'retail' mode
2. search for AtlasLootClassic
3. install
4. fail

**Comment by torkus** [comment]
problem lies here: https://github.com/ogri-la/wowman/blob/develop/src/wowman/curseforge_api.clj#L58-L64

and it poses an interesting question: 

> if your addon directory is "retail" and you want to install an addon that is classic *only*, what should wowman do?

* refuse to install addon?
* install the classic version and then issue warnings on each refresh?
* prompt user to confirm they're about to install a classic-only addon into a retail mode directory?
* hide classic-only search results so they can't be installed?

any suggestions @layday?

**Comment by layday** [comment]
I suppose I would begin by asking if there is a use case for installing classic CurseForge add-ons to retail; if there is one, it must be very obscure.  My own approach towards things like this is to follow the path of - shall we say - least programmatic resistance.  I wouldn't go into the effort of special-casing installing and updating add-ons from a different 'track' unless there is some sort of demand or traction.

**Comment by torkus** [comment]
(fix here: https://github.com/ogri-la/wowman/pull/56/files)

I hear what you're saying, thank you for the feedback. 

The fix is the minimal amount to get this working again and essentially refuses to install the addon with a warning. Re-installing an addon for the wrong game track fails similarly. In 0.11.0 I might investigate a friendlier approach

**Comment by torkus** [comment]
released in 0.9.2
